src/item_box.py

```diff
@@ -57,52 +60,42 @@ def add_actions(self):
         # Set the action group for the widget
         for name in action_list:
             action = Gio.SimpleAction.new(name, None)
-            action.connect(
-                "activate", getattr(self, f"{name}"), self.item,
-            )
+            action.connect("activate", getattr(self, f"{name}"))
             action_group.add_action(action)
         self.insert_action_group("item_box", action_group)
 
+    def _change_position(self, source_index, target_index):
+        application = self.get_application()
+        application.get_data().change_position(target_index, source_index)
+        clipboard = application.get_clipboard()
+        clipboard.append((3, (source_index, target_index)))
+        clipboard.add()
+        application.get_view_clipboard().add()
+
     def on_dnd_drop(self, drop_target, value, _x, _y):
         # Handle the dropped data here
-        data = self.get_application().get_data()
-        before_index = data.index(value)
-        data.change_position(drop_target.uuid, value)
-        clipboard = self.get_application().get_clipboard()
-        clipboard.append((3, (before_index, data.index(value))))
-        clipboard.add()
-        self.get_application().get_view_clipboard().add()
+        self._change_position(int(value), int(drop_target.index))
 
     def on_dnd_prepare(self, drag_source, x, y):
         snapshot = Gtk.Snapshot.new()
         self.do_snapshot(self, snapshot)
         paintable = snapshot.to_paintable()
         drag_source.set_icon(paintable, int(x), int(y))
-        return Gdk.ContentProvider.new_for_value(self.props.item.uuid)
+        return Gdk.ContentProvider.new_for_value(str(self.props.index))
 
     def on_color_change(self, item, _ignored):
         self.provider.load_from_data(
-            f"button {{ color: {item.color}; opacity: {item.alpha};}}", -1)
-
-    def move_up(self, _action, _shortcut, item):
-        data = self.get_application().get_data()
-        before_index = data.index(item)
-
-        data.change_position(item.uuid, data[before_index - 1].uuid)
-        clipboard = self.get_application().get_clipboard()
-        clipboard.append((3, (before_index, data.index(item))))
-        clipboard.add()
-        self.get_application().get_view_clipboard().add()
+            "button { "
+            f"color: {item.get_color()}; "
+            f"opacity: {item.get_alpha()}; "
+            "}", -1,
+        )
 
-    def move_down(self, _action, _shortcut, item):
-        data = self.get_application().get_data()
-        before_index = data.index(item)
+    def move_up(self, _action, _shortcut):
```

Comment:
This was a refactor I had in my mind as well, given the strong resemblence between these two functions